tools/Repo-Tasks.psm1

```diff
@@ -0,0 +1,253 @@
+This will enable to execute Start-RepoBuild <scope>
+
+#>
+
+    Write-Host "Below are available scopes you can specify for building specific projects"
+    Write-Host ""    
+    Get-ChildItem -path "$env:repoRoot\src\ResourceManager" -dir | Format-Wide -Column 5 | Format-Table -Property Name    
+    Get-ChildItem -path "$env:repoRoot\src\ServiceManagement" -dir | Format-Wide -Column 5 | Format-Table -Property Name
+}
+
+[CmdletBinding]
+Function Start-RepoBuild
+{
+<#
+.SYNOPSIS
+This cmdlet will help to do either with full build or targeted build for specific scopes.
+
+.PARAMETER BuildScope
+Use Get-BuildScope cmdLet to get list of existing scopes that can be used to build
+#>
+    param(
+    [parameter(Mandatory=$false, Position=0, HelpMessage='BuildScope that you would like to use. For list of build scopes, run List-BuildScopes')]
+    [string]$BuildScope
+    )    
+    
+    if([string]::IsNullOrEmpty($BuildScope) -eq $true)
+    {
+       Write-Host "Starting Full build"
+       msbuild.exe "$env:repoRoot\build.proj" /t:Build
+    }
+    else
+    {
+        Write-Host "Building $BuildScope"
+        msbuild.exe "$env:repoRoot\build.proj" /t:Build /p:Scope=$BuildScope
+    }
+}
+
+[CmdletBinding]
+Function Invoke-CheckinTests
+{
+<#
+.SYNOPSIS
+Runs all the check in tests
+#>
+    Write-Host "cmdline Args: msbuild.exe $env:repoRoot\build.proj /t:Test"
+    msbuild.exe "$env:repoRoot\build.proj" /t:Test
+}
+
+[cmdletBinding]
+Function Install-VSProjectTemplates
+{
+<#
+.SYNOPSIS
+
+Install-VSProjectTemplates will install getting started project templates for
+1) Autorest-.NET SDKProject
+2) .NET SDK Test project
+3) PowerShell Test Project
+
+After executing the cmdlet, restart VS (if already open), create new project
+Search for the project template as we install the following three project templates
+AutoRest-AzureDotNetSDK
+AzureDotNetSDK-TestProject
+AzurePowerShell-TestProject
+#>
+    if($env:VisualStudioVersion -eq "14.0")
+    {
+        if((Test-Path "$env:repoRoot\tools\ProjectTemplates\") -eq $true)
+        {
+            Write-Host "Installing VS templates for 'AutoRest as well as Test Project'"
+            Copy-Item "$env:repoRoot\tools\ProjectTemplates\*.zip" "$env:USERPROFILE\Documents\Visual Studio 2015\Templates\ProjectTemplates\"
+            Write-Host "Installed VS Test Project Templates for Powershell and DotNet SDK test projects"
+            Write-Host "Restart VS (if already open), search for 'AutoRest-AzureDotNetSdk', 'AzureDotNetSDK-TestProject' or 'AzurePowerShell-TestProject'"
+        }
+        else
+        {
+            Write-Host "Missing templates to install, make sure you have project templates available in the repo under $env:repoRoot\tools\ProjectTemplates\"
+        }
+    }
+    else
+    {
+        Write-Host "Unsupported VS Version detected. Visual Studio 2015 is the only supported version for current set of project templates"
+    }
+}
+
+Function Create-ServicePrincipal-Help()
```

Comment:
Let's remove until this actually creates a service principal.  Once it does, we will need to add a module manifest, as we will have some dependencies.